execution/streaming-partitioner/src/main.rs

```diff
@@ -0,0 +1,192 @@
+            cut_edges_weight as f64 / total_edges_weight as f64
+        );
+    }
+}
+
+fn main() {
+    println!("Starting the transaction orderer benchmark");
+    let args = Args::parse();
+    let num_accounts = args.num_accounts;
+    println!("Creating {} accounts", num_accounts);
+    let accounts: Vec<Mutex<TestAccount>> = (0..num_accounts)
+        .into_par_iter()
+        .map(|_i| Mutex::new(generate_test_account()))
+        .collect();
+    println!("Created {} accounts", num_accounts);
+    println!("Creating {} transactions", args.block_size);
+    let transactions: Vec<AnalyzedTransaction> = (0..args.block_size)
+        .map(|_| {
+            // randomly select a sender and receiver from accounts
+            let mut rng = OsRng;
+
+            let indices = rand::seq::index::sample(&mut rng, num_accounts, 2);
+            let receiver = accounts[indices.index(1)].lock().unwrap();
+            let mut sender = accounts[indices.index(0)].lock().unwrap();
+            create_signed_p2p_transaction(&mut sender, vec![&receiver]).remove(0)
+        })
+        .collect();
+
+    let now = Instant::now();
+    let transactions = compress_transactions(transactions);
+    println!("Mapping time: {:?}", now.elapsed());
+    let transactions = transactions.into_iter().batched(args.block_size);
+    let x = |_: &CompressedPTransaction<AnalyzedTransaction>| 1 as NodeWeight;
+    let mut params = transaction_graph_partitioner::Params {
+        node_weight_function: x,
+        edge_weight_function,
+        shuffle_batches: false,
+    };
+
+    match args.partitioner {
+        Partitioner::Fennel => {
+            let mut fennel = FennelGraphPartitioner::new(args.num_shards);
+            fennel.balance_constraint_mode = BalanceConstraintMode::Batched;
+            fennel.alpha_computation_mode = AlphaComputationMode::Batched;
+            params.shuffle_batches = true;
+            run_benchmark(
+                transactions,
+                TransactionGraphPartitioner::new(fennel, params),
+                args,
+            )
+        },
+        #[cfg(feature = "metis-partitioner")]
+        Partitioner::Metis => {
+            let metis = MetisGraphPartitioner::new(args.num_shards);
+            let metis_streaming = WholeGraphStreamingPartitioner::new(metis);
+            run_benchmark(
+                transactions,
+                TransactionGraphPartitioner::new(metis_streaming, params),
+                args,
+            )
+        },
+        Partitioner::Random => {
+            let random_partitioner = RandomPartitioner::new(args.num_shards);
+            run_benchmark(
+                transactions,
+                TransactionGraphPartitioner::new(random_partitioner, params),
+                args,
+            )
+        },
+    };
+}
+
+pub fn edge_weight_function(idx1: SerializationIdx, idx2: SerializationIdx) -> EdgeWeight {
+    ((1. / (1. + idx1 as f64 - idx2 as f64)) * 100000.) as EdgeWeight
```

Comment:
@xosmig how was 100000 chosen? is the ratio between this value and the default node weight matter?

Reply:
It was pretty arbitrary. This whole edge weight function is not very scientific and was just the first thing I came up with. 10^5 just seemed like enough precision for the tests I was running. However, on some inputs, it caused overflows.

I imagine, eventually, we will use our own partitioner implementation and will be able to use floats instead of i32, so there won't be such problems.